lib/styles.js

```diff
@@ -340,4 +342,145 @@ module.exports = function (expect) {
+        if (markUpSpecialCharacters) {
+            text.split(specialCharRegExp).forEach(function (part) {
+                if (specialCharRegExp.test(part)) {
+                    this[specialCharStyle || baseStyle](utils.escapeChar(part));
+                } else {
+                    this[baseStyle](part);
+                }
+            }, this);
+        } else {
+            this[baseStyle](text);
+        }
+    });
+
+    expect.addStyle('stringDiff', function (actual, expected, options) {
+        options = options || {};
+        var type = options.type || 'WordsWithSpace';
+        var diffLines = [];
+        var lastPart;
+        stringDiff.diffLines(actual, expected).forEach(function (part) {
+            if (lastPart && lastPart.added && part.removed) {
+                diffLines.push({
+                    oldValue: part.value,
+                    newValue: lastPart.value,
+                    replaced: true
+                });
+                lastPart = null;
+            } else if (lastPart) {
+                diffLines.push(lastPart);
+                lastPart = part;
+            } else {
+                lastPart = part;
+            }
+        });
+        if (lastPart) {
+            diffLines.push(lastPart);
+        }
+
+        diffLines.forEach(function (part, index) {
+            var endsWithNewline = /\n$/.test(part.value);
+            var value;
+            if (part.replaced) {
+                var oldLine = this.clone();
+                var newLine = this.clone();
+                var oldValue = part.oldValue;
+                var newValue = part.newValue;
+                var oldEndsWithNewline = oldValue.slice(-1) === '\n';
+                var newEndsWithNewline = newValue.slice(-1) === '\n';
+                if (oldEndsWithNewline) {
+                    oldValue = oldValue.slice(0, -1);
+                }
+                if (newEndsWithNewline) {
+                    newValue = newValue.slice(0, -1);
+                }
+
+                stringDiff['diff' + type](oldValue, newValue).forEach(function (part) {
+                    if (part.added) {
+                        newLine.diffFragment(part.value, 'diffAddedHighlight', 'diffAddedSpecialChar', options.markUpSpecialCharacters);
+                    } else if (part.removed) {
+                        oldLine.diffFragment(part.value, 'diffRemovedHighlight', 'diffRemovedSpecialChar', options.markUpSpecialCharacters);
+                    } else {
+                        newLine.diffAddedLine(part.value);
+                        oldLine.diffRemovedLine(part.value);
+                    }
+                });
+                this.alt({
+                    text: function () {
+                        oldLine.prependLinesWith(this.clone().diffRemovedLine('-'));
+                        newLine.prependLinesWith(this.clone().diffAddedLine('+'));
+                    },
+                    fallback: function () {}
```

Comment:
Here it would be nice of `magicPen.alt` allowed omitting `fallback: function () {}`.